src/main/java/gg/uhc/uhc/modules/health/HealthCommand.java

```diff
@@ -48,39 +52,105 @@
 public class HealthCommand extends TemplatedOptionCommand {
 
     protected static final NumberFormat FORMAT = NumberFormat.getInstance();
+    protected static final String MODIFY_PERMISSION = "uhc.command.health.modify";
 
     static {
         FORMAT.setMaximumFractionDigits(2);
         FORMAT.setMinimumFractionDigits(0);
     }
 
     protected final OptionSpec<Player> playerSpec;
+    protected final OptionSpec<Void> allPlayersSpec;
+    protected final OptionSpec<Double> setCurrentHealthSpec;
+    protected final OptionSpec<Double> setMaximumHealthSpec;
+    protected final double maximumHealthLimit;
 
-    public HealthCommand(MessageTemplates messages) {
+    public HealthCommand(MessageTemplates messages, double maximumHealthLimit) {
         super(messages);
+        this.maximumHealthLimit = maximumHealthLimit;
 
-        playerSpec = parser.nonOptions("List of player names to check the health of")
+        playerSpec = parser.nonOptions("List of player names to check/modify the health of")
                 .withValuesConvertedBy(new OnlinePlayerConverter());
 
         nonOptionsTabComplete = new NonDuplicateTabComplete(OnlinePlayerTabComplete.INSTANCE);
+
+        allPlayersSpec = parser
+                .acceptsAll(ImmutableList.of("a", "all"), "Check/modify all online players");
+
+        setCurrentHealthSpec = parser
+                .accepts("set", "Set the health of players")
+                .withRequiredArg()
+                .withValuesConvertedBy(new DoubleConverter()
+                        .setPredicate(DoublePredicates.GREATER_THAN_ZERO)
+                        .setType("New player health")
+                );
+
+        setMaximumHealthSpec = parser
+                .accepts("setMaximum", "Set the maximum health of players")
+                .withRequiredArg()
+                .withValuesConvertedBy(new DoubleConverter()
+                        .setPredicate(DoublePredicates.GREATER_THAN_ZERO)
+                        .setType("New maximum player health")
+                );
```

Comment:
I like to always give each option a short flag as an alternative to the long flag. Maybe -m and -s for these two? If you could add some example usages of the command to the doc.md that would be good too. `/h -am 40 -s 40` wouldn't be too hard to remember for 20 hearts games. I'm still not 100% sure on the mixing of /h with a set health mechanic tbh but it seems like it should work OK